src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+    private val securableObjectTypes: IMap<AclKey, SecurableObjectType> = hazelcastInstance.getMap(HazelcastMap.SECURABLE_OBJECT_TYPES.name)
+    private val aces: IMap<AceKey, AceValue> = hazelcastInstance.getMap(HazelcastMap.PERMISSIONS.name)
+    private val logger = LoggerFactory.getLogger(ExternalDatabaseManagementService::class.java)
+
+    /*CREATE*/
+    fun createOrganizationExternalDatabaseTable(orgId: UUID, table: OrganizationExternalDatabaseTable): UUID {
+        val principal = Principals.getCurrentUser()
+        Principals.ensureUser(principal)
+
+        val tableFQN = FullQualifiedName(orgId.toString(), table.name)
+        checkState(organizationExternalDatabaseTables.putIfAbsent(table.id, table) == null,
+                "OrganizationExternalDatabaseTable ${tableFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(table, tableFQN::getFullQualifiedNameAsString)
+
+        val tableAclKey = AclKey(orgId, table.id)
+        authorizationManager.setSecurableObjectType(tableAclKey, SecurableObjectType.OrganizationExternalDatabaseTable)
+        authorizationManager.addPermission(tableAclKey, principal, EnumSet.allOf(Permission::class.java))
+        //eventBus?
+
+        return table.id
+    }
+
+    fun createOrganizationExternalDatabaseColumn(orgId: UUID, column: OrganizationExternalDatabaseColumn): UUID {
+        val principal = Principals.getCurrentUser()
+        Principals.ensureUser(principal)
+
+        checkState(organizationExternalDatabaseTables[column.tableId] != null,
+                "OrganizationExternalDatabaseColumn ${column.name} belongs to a table that does not exist")
+        val columnFQN = FullQualifiedName(column.tableId.toString(), column.name)
+        checkState(organizationExternalDatabaseColumns.putIfAbsent(column.id, column) == null,
+                "OrganizationExternalDatabaseColumn ${columnFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(column, columnFQN::getFullQualifiedNameAsString)
+
+        val columnAclKey = AclKey(orgId, column.tableId, column.id)
+        authorizationManager.setSecurableObjectType(columnAclKey, SecurableObjectType.OrganizationExternalDatabaseColumn)
+        authorizationManager.addPermission(columnAclKey, principal, EnumSet.allOf(Permission::class.java))
+
+        return column.id
+    }
+
+    /*GET*/
+    fun getOrganizationIds(): Set<UUID> {
+        val orgIds = mutableSetOf<UUID>()
+        hds.connection.createStatement().use {
+            val rs = it.executeQuery("SELECT id FROM organizations")
+            while (rs.next()) {
+                orgIds.add(organizationId(rs))
+            }
+        }
+        return orgIds
+    }
+
+    fun getOrganizationExternalDatabaseTable(tableId: UUID): OrganizationExternalDatabaseTable {
+        return organizationExternalDatabaseTables[tableId]!!
```

Comment:
Also super minor, but for consistency, I think we've been opting for the `a.getValue(b)` syntax over the `a[b]!!` syntax

Reply:
Gotcha. Kotlin told me to change it but I'll change it back.